Homework 23 Selenium Page Factory Revised 2024-06-23

jmar138 opened the pull request:
Homework 23 Selenium Page Factory Revised 2024-06-23

jmar138 committed:
Update LoginTests.java

jmar138 committed:
Merge pull request #1 from jmar138/HomeWork-15

Update LoginTests.java

TestPro-Support committed:
Added L16 examples

TestPro-Support committed:
Updated email domain

testpro-student committed:
Refactoring of Login test cases, added Change name test

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

Homework-17

jmar138 committed:
Homework-17

Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Merge branch 'main' into Homework-17

jmar138 committed:
Homework-17

Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-19

Homework-19

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-20

Homework-20 Revised

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-22

Homework-22

jmar138 committed:
Homework-17

jmar138 committed:
Homework-23

Homework-23 Revised

jmar138 committed:
Homework-17

jmar138 committed:
Homework-22 Revised

Homework-22 Revised

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-17

jmar138 committed:
Homework-23 - PageFactory

Homework-23 - PageFactory:

Revised - 2024-06-23

jmar138 committed:
Homework-17

Berichicko commented:
The code is not compiled! Please finish the homework!You got this problem, please correct it.

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/dbf1fc1c-3fa2-448c-bc2d-d1c953395667)

**HINT**

**Step 1**

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/861ffbe8-63d7-44c9-85cd-3ab3a46c1aa8)

**Step 2**

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/ccad4d21-927c-46ff-9c39-db1f33743e32)

Berichicko commented:
```
 public static WebDriver pickBrowser(String browser) throws MalformedURLException {
        DesiredCapabilities capabilities = new DesiredCapabilities();
         String gridURL = "http://192.168.55.103:4444";  //replace with your gird url,  localhost:4444
        switch(browser) {
            case "firefox":  // gradle clean test -Dbrowser=firefox
                WebDriverManager.firefoxdriver().setup();
                FirefoxOptions firefoxOptions = new FirefoxOptions();
                firefoxOptions.addArguments("--no-notifications");
                return driver = new FirefoxDriver();
            case "edge":   // gradle clean test -Dbrowser=edge
                WebDriverManager.edgedriver().setup();
                EdgeOptions edgeOptions = new EdgeOptions();
                edgeOptions.addArguments("--remote-allow-origins=*");   // Allow automations to be run
                edgeOptions.addArguments("--disable-notifications");    // disabling notifications
                edgeOptions.addArguments("--inprivate");                // launching in incognito mode
                edgeOptions.addArguments("--window-position=250,0");    // move the window over to the right
                return driver = new EdgeDriver();
            case "grid-edge": // gradle clean test -Dbrowser=grid-edge
                capabilities.setCapability("browserName", "edge");
               return driver = driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
            case "grid-firefox": // gradle clean test - Dbrowser=grid-firefox
                capabilities.setCapability("browserName", "firefox");
                return driver = driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
            case "grid-chrome":  // gradle clean test - Dbrowser=grid-chrome
                capabilities.setCapability("browserName", "chrome");
                return driver = driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
            default:
                WebDriverManager.chromedriver().setup();
                ChromeOptions chromeOptions = new ChromeOptions();
                chromeOptions.addArguments("--Remote-allow-origins=*");
                chromeOptions.addArguments("--disable-notifications");    // disabling notifications
                chromeOptions.addArguments("--incognito");                // launching in incognito mode
                chromeOptions.addArguments("--window-position=250,0");    // move the window over to the right
                return driver = new ChromeDriver(chromeOptions);
        }
    }
```

jmar138 committed:
Homework-23 Selenium Page Factory

Homework-23 Selenium Page Factory
Revised: 2024-06-25

Fails and the error I get is "java.lang.ClassCastException: class jdk.proxy2.$Proxy22 cannot be cast to class org.openqa.selenium.By (jdk.proxy2.$Proxy22 is in module jdk.proxy2 of loader 'app'; org.openqa.selenium.By is in unnamed module of loader 'app')"
